android/KMEA/app/src/main/java/com/tavultesoft/kmea/util/BCP47.java

```diff
@@ -18,4 +20,36 @@ public static boolean languageEquals(String id1, String id2) {
 
     return id1.equalsIgnoreCase(id2);
   }
-}
\ No newline at end of file
+
+  /**
+   * Utility to modify languageList.
+   * If languageID exists in the list, remove it. Otherwise, add languageID to the list.
+   * @param languageList
+   * @param languageID
+   */
+  public static void toggleLanguage(ArrayList<String> languageList, String languageID) {
+    if (languageList != null) {
+      if (languageList.size() == 0) {
+        languageList.add(languageID.toLowerCase());
+        return;
+      }
```

Comment:
```suggestion
```

This is additional logic which is not needed -- the loop will be a no-op in this case and we'll do the add immediately afterwards.